modules/diskq/logqueue-disk.c

```diff
@@ -289,3 +289,22 @@ log_queue_disk_serialize_msg(LogQueueDisk *self, LogMessage *msg, GString *seria
   serialize_archive_free(sa);
   return error == NULL;
 }
+
+gboolean
+log_queue_disk_deserialize_msg(LogQueueDisk *self, GString *serialized, LogMessage **msg)
+{
+  SerializeArchive *sa = serialize_string_archive_new(serialized);
+  LogMessage *local_msg = log_msg_new_empty();
+
+  if (!log_msg_deserialize(local_msg, sa))
+    {
+      log_msg_unref(local_msg);
```

Comment:
The error message `Error deserializing message from the disk-queue file` is missing from here.